### ISSUE TITLE
Rewind radioi broadcasts

### ISSUE THREAD

**Issue by sigjak** [ryanheise/just_audio#595]
**Is your feature request related to a problem? Please describe.**
Sometimes I would like to be able to rewind when listening to streaming radio broadcasts.

**Describe the solution you'd like**
I would like to suggest something that may already be available or possible but has escaped me. If so, I apologize beforehand. 
It would be nice to have method in just_audio  which would make it possible to to go backwards in a continuous  stream such as a radio station  broadcast stream and start listening some time before ‘now’.  Something similar to what player.seek() does for fixed duration streams.  maybe player.rewind(). Would this be feasible?

All the best
Sig

**Describe alternatives you've considered**
No solution has come to mind

**Additional context**

**Comment by ryanheise** [comment]
I agree it would be useful to not only seek back in a livestream, but also seek forward to return to the tip of the livestream (the latter has already been requested in a previous feature request).

I have experimented with this a bit in the past but didn't have any success on iOS. If iOS programmers would like to have a go, I welcome pull requests.

In terms of what the API should look like, I think it should go through the standard `seek` method but maybe there needs to be an additional API to allow you to query what is the current seekable range.

**Comment by swiftymf** [comment]
I think this depends on the type of url/stream that's being used. I am using an HLS stream for one of my live radio broadcasts and am able to rewind some amount using AudioPlayer.position. I set the button to rewind 10 seconds each time is tapped. I know with my HLS stream I am getting about 120 seconds of buffer(?) and I think that's what allows me to rewind. 

There is another issue around caching that could possibly help solve this? I'm trying to learn more about how that works here and/or using flutter_cache_manager to try to set something up for my project.

**Comment by ryanheise** [comment]
There are also some buffering options that can be passed to the constructor of `AudioPlayer` but on iOS I don't see any option for configuring the backwards buffer. For these sorts of things, I would typically search on how native iOS programmers solve this type of use case and then try to replicate the equivalent in this plugin.

**Comment by swiftymf** [comment]
I set this up to try out what I mentioned. 

- Start downloading the audio (let it download for 10 seconds before playing for test purposes)
- Play audio from file path
- Audio pauses around 18 seconds, which appears to be how much was downloaded when we started playing the audio
- If I reduce the download time before Play then the duration that plays is less too which is why I believe the above is true.

The audio continues to download, but it seems like just_audio doesn't know about anything downloaded after play started. 

` 
  @override
  Future<void> playMediaItem(MediaItem mediaItem) async {
    await _audioPlayer.dispose();
    _setupAudioPlayer();
    updateMediaItem(mediaItem);

    var dir = await getApplicationDocumentsDirectory();
    var fullPath = dir.path + '/Audios/audio.mp3';

    // Delete previous file if it exists
    final file = File(fullPath);
    if (file.existsSync()) {
      file.delete();
    }

    // start downloading the audio
    // Allow to download for 10 seconds before playing audio
    await _downloadRadio(Uri.parse(mediaItem.id));

    try {
      await _audioPlayer.setAsset(fullPath)
      _audioPlayer.play();

    } catch (e) {
      Fimber.d('_audioPlayer.play() error: $e');
    }
  }

 _downloadRadio(Uri uri) async {
    final dio = Dio();
    var dir = await getApplicationDocumentsDirectory();
    var fullPath = dir.path + '/Audios/audio.mp3';
    dio.downloadUri(uri, fullPath, onReceiveProgress: (received, total) {
      log('dio progress: $received, $total');
    });
    // allow download for 10 seconds
    await futureWait(milliseconds: 10000);
  }
`

**Comment by ryanheise** [comment]
I'm not sure if the file idea will work. I would still suggesting to search how native iOS developers have solved this problem in the past and then try to recreate that.
